ci-operator/step-registry/nmstate/bond/nmstate-bond-commands.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function check_pods_status() {
+  POD_STATUSES=$(oc get pods -n openshift-nmstate --no-headers -o custom-columns=:status.phase)
+  ALL_RUNNING=true
+
+  for STATUS in $POD_STATUSES; do
+    if [ "$STATUS" != "Running" ]; then
+      ALL_RUNNING=false
+      break
+    fi
+  done
+
+  echo $ALL_RUNNING
+}
+
+function deploy_nmstate_operator(){
+    cat <<EOF | oc create -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    kubernetes.io/metadata.name: openshift-nmstate
+    name: openshift-nmstate
+  name: openshift-nmstate
+spec:
+  finalizers:
+  - kubernetes
+---
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  annotations:
+    olm.providedAPIs: NMState.v1.nmstate.io
+  generateName: openshift-nmstate-
+  name: kubernetes-openshift-nmstate-dzrmx
+  namespace: openshift-nmstate
+spec:
+  targetNamespaces:
+  - openshift-nmstate
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  labels:
+    operators.coreos.com/kubernetes-nmstate-operator.openshift-nmstate: ""
+  name: nmstate-operator-sub
+  namespace: openshift-nmstate
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: kubernetes-nmstate-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+EOF
+    if [ $? == 0 ]; then
+        echo "create nmstate operator successfully" 
+    else
+        echo "!!! fail to create nmstate operator "
+        return 1
+    fi
+    sleep 60
+
+    cat <<EOF | oc create -f -
+    apiVersion: nmstate.io/v1
+    kind: NMState
+    metadata:
+      namespace: openshift-nmstate
+      name: nmstate
+EOF
+    if [ $? == 0 ]; then
+        echo "create nmstate config successfully" 
+    else
+        echo "!!! fail to create nmstate config"
+        return 1
+    fi
+    sleep 60
+    oc get pod -n openshift-nmstate
+}
+
+echo "Now begin to set the bond by nmstate operator"
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+currentPlugin=$(oc get network.config.openshift.io cluster -o jsonpath='{.status.networkType}')
+if [ ${currentPlugin} != "OpenShiftSDN" ]; then
+  echo "Exiting script because ovn cannot set the default network with nmstate operator nncp"
+  exit
+fi
+
+default_iface=$(oc debug node/"$(oc get node -lnode-role.kubernetes.io/master="" -o jsonpath='{.items[0].metadata.name}')" -n default -- chroot /host/ bash -c 'ip route' | grep default | head -1 | awk '{print $5}')
```

Comment:
One question here, do all nodes have the same default interface?

Reply:
yes. by default is ens192 on vsphere